### ISSUE TITLE
Pre-training procedure for ELECTRA

### ISSUE THREAD

**Issue by ameet-1997** [jcblaisecruz02/Filipino-Text-Benchmarks#1]
Would it possible to share details about how the ELECTRA models were pre-trained?
What code base was used for the same? (Modified HuggingFace, or the Google Research original implementation?)

**Comment by jcblaisecruz02** [comment]
Hi! Sorry the details aren't released yet as we're still writing the paper associated with these models.

But yes, we used Google Research's implementation, we only use HuggingFace Transformers for testing and deployment. The pretraining recipe also follows the one in the paper, ie:

**ELECTRA small** 
- learning rate 5e-4
- batch size 128
- msl 128
- generator hidden size 25% of discriminator's hidden size
- trained for 1M steps

**ELECTRA Base**
- learning rate 2e-4
- batch size 256
- msl 512
- generator hidden size 33% of discriminator's hidden size
- trained for 766K steps

Both models use a vocabulary size of 32k generated from WikiText-TL-39.
